### ISSUE TITLE
Release 1.11

### ISSUE THREAD

**Issue by kfunaoka** [autowarefoundation/autoware_ai#476]

**Comment by kfunaoka** [comment]
I'd like to add autowarefoundation/autoware_ai#474 and autowarefoundation/autoware_ai#475 for more maintainability.
autowarefoundation/autoware_ai#475 can delete a part of obsolete ones (not all of them).

**Comment by mitsudome-r** [comment]
I'd like to add autowarefoundation/autoware#1817 as Bug Fix

**Comment by kfunaoka** [comment]
@mitsudome-r It should be! I've added autowarefoundation/autoware#1817 into the headline.

I'd like to add autowarefoundation/autoware#1609. It's too huge and should be well organized. @s-azumi, @mitsudome-r, and me will take care of it.

**Comment by amc-nu** [comment]
Adding autowarefoundation/autoware#1800 autowarefoundation/autoware#1645 and autowarefoundation/autoware_ai#464

**Comment by k0suke-murakami** [comment]
Hope to add autowarefoundation/autoware_ai#452 autowarefoundation/autoware_ai#265

**Comment by sgermanserrano** [comment]
Adding autowarefoundation/autoware#1594

**Comment by hakuturu583** [comment]
Adding autowarefoundation/autoware#1848

**Comment by isamu-takagi** [comment]
Adding autowarefoundation/autoware_ai#375

**Comment by hakuturu583** [comment]
Adding autowarefoundation/autoware_ai#494

**Comment by hakuturu583** [comment]
Adding autowarefoundation/autoware#1893

**Comment by kfunaoka** [comment]
@kitsukawa requested to add autowarefoundation/autoware_ai#510

**Comment by YamatoAndo** [comment]
Adding autowarefoundation/autoware_ai#460

**Comment by YamatoAndo** [comment]
Adding autowarefoundation/autoware_ai#506 autowarefoundation/autoware_ai#517

**Comment by kitsukawa** [comment]
Adding autowarefoundation/autoware_ai#462

**Comment by mitsudome-r** [comment]
add autowarefoundation/autoware_ai#499

**Comment by yn-mrse** [comment]
Adding autowarefoundation/autoware_ai#272 autowarefoundation/autoware_ai#274 autowarefoundation/autoware_ai#532

**Comment by k0suke-murakami** [comment]
Adding autowarefoundation/autoware_ai#534

**Comment by mitsudome-r** [comment]
Adding autowarefoundation/autoware_ai#541

**Comment by hakuturu583** [comment]
Adding autowarefoundation/autoware#1943

**Comment by amc-nu** [comment]
Adding autowarefoundation/autoware#1966

**Comment by hakuturu583** [comment]
Adding autowarefoundation/autoware#1970

**Comment by filiperinaldi** [comment]
Can autowarefoundation/autoware#1946 be included please.

**Comment by hakuturu583** [comment]
Adding autowarefoundation/autoware_ai#566 ,#2005

**Comment by s-azumi** [comment]
Adding autowarefoundation/autoware_ai#564

**Comment by hakuturu583** [comment]
Adding autowarefoundation/autoware#2012

**Comment by sgermanserrano** [comment]
The following is an overview of the current state of PRs for new features that are meant to be included in v1.11. Taking into account the amount of work that is still needed for some of them and the limited time for v1.11, can the current developers make a decision as to whether the PRs will be ready in time for appropriate review?

| No. |  PR / Issue | Developer | Approver | Notes |
| :----- | :----- | :-----: | :-----: | :----- | 
| 1 | Feature/rebuild decision maker autowarefoundation/autoware#1609 autowarefoundation/autoware_ai#564 | @kfunaoka @s-azumi  |  N/A |  Tests under development for the different packages of this PR in autowarefoundation/autoware#1757 autowarefoundation/autoware#1744 autowarefoundation/autoware#1958 autowarefoundation/autoware#2009 | 
| 2 | Initial release of Shape Estimation autowarefoundation/autoware#1815 | @yukkysaito @amc-nu |  @amc-nu |  Unit tests were added as part of the PR | 
| 3 | Initial release of Baidu's LiDAR based CNN segmentation autowarefoundation/autoware#1800 | @amc-nu |  @cirpue49 |  **NO tests** were included for this feature prior to merging |  
| 4 | Initial release of Object Filter autowarefoundation/autoware#1645 | @amc-nu |  N/A |  Closed and moved to autowarefoundation/autoware#1815 | 
| 5 | Initial release of FireSSD image based detector for CPU autowarefoundation/autoware_ai#464 | @amc-nu |  N/A |  No associated PR for this issue |  
| 6 | Improve Hybrid A* planner autowarefoundation/autoware#1594  | @aohsato |  N/A | Moved to 1.11 from 1.10. Some tests added in autowarefoundation/autoware#1753, still missing tests for freespace_planner and astar_avoid | 
| 7 | Initial release of Autoware Launcher autowarefoundation/autoware_ai#375 autowarefoundation/autoware#1950 | @isamu-takagi |  @hakuturu583 |  Some tests added as part of [commit](https://github.com/CPFL/Autoware/pull/1950/commits/c4f1b76e74570a969f11eb5de3c3e04131a214a1) | 
| 8 | Set negative esimate_twist.twist.linear.x if going backwards autowarefoundation/autoware_ai#460 then autowarefoundation/autoware#1798 | @YamatoAndo |  @kitsukawa |  Merged with *no test* added for new method |  
| 9 | Can not reach the end of waypoint autowarefoundation/autoware_ai#272 and autowarefoundation/autoware#1477  | @yn-mrse |  N/A |  No tests added yet | 
| 10 | Simplified configuration of waypoint functions autowarefoundation/autoware_ai#274 and autowarefoundation/autoware#1951 | @yn-mrse |  @hakuturu583 |  **NO tests** were included for this feature prior to merging | 
| 11 | Enable backward driving autowarefoundation/autoware#1940 | @yn-mrse |  N/A |  No associated PR for this issue  | 
| 12 | [Feature] costmap_generator autowarefoundation/autoware_ai#452 and autowarefoundation/autoware#1774 | @cirpue49 |  N/A |  Tests added, awaiting review  | 
| 13 | [Feature] Robust Adavitive Uncented Kalman Filter + IMM-UKF-PDA autowarefoundation/autoware_ai#265 and autowarefoundation/autoware#1685 | @cirpue49 |  N/A |  Tests added, awaiting review  | 
| 14 | Feature/vehicle description autowarefoundation/autoware#1848 | @hakuturu583 |  @yukkysaito |  **NO tests** were included for this feature prior to merging  | 
| 15 | Feature/lgsvl sim configurator autowarefoundation/autoware#1893 | @hakuturu583  |  @yn-mrse |  **NO tests** were included for this feature prior to merging  | 
| 16 | [Features] Mobileeye Driver autowarefoundation/autoware_ai#510 | @kitsukawa |  N/A |  No associated PR for this issue  | 
| 17 | Feature/RosbagController autowarefoundation/autoware#1791 | @ChenxiTU @kitsukawa |  @isamu-takagi |  **NO tests** were included for this feature prior to merging  | 
| 18 | [Feature] feature/nmea_to_mgrs autowarefoundation/autoware_ai#499 and autowarefoundation/autoware#1878 | @mitsudome-r |  N/A |  Closed  | 
| 19 | [Feature] Gazebo simulator autowarefoundation/autoware_ai#512 and autowarefoundation/autoware#1930  | @yukkysaito | @aohsato|  **NO tests** were included for this feature prior to merging   | 
| 20 | [Feature] Point Pillar for 3D bounding box detection from LIDAR data autowarefoundation/autoware_ai#534 | @cirpue49 |  N/A |   No associated PR for this issue   | 
| 21 | [Feature] Load other map formats autowarefoundation/autoware_ai#541 | @mitsudome-r |  N/A |  No associated PR for this issue  | 
| 22 | Feature/autoware health checker autowarefoundation/autoware#1943, related to autowarefoundation/autoware#1970, autowarefoundation/autoware#2003, autowarefoundation/autoware#2004, autowarefoundation/autoware_ai#567, autowarefoundation/autoware#2006 | @hakuturu583 |  @yn-mrse |  Some tests added. Bug reported on autowarefoundation/autoware#2012 |

**Comment by YamatoAndo** [comment]
Adding  autowarefoundation/autoware_ai#225

**Comment by hakuturu583** [comment]
Sorry for my late...
This is fix commit for ./run script.
autowarefoundation/autoware_ai#585 
I want to add thie PR to the ver 1.11

**Comment by isamu-takagi** [comment]
Adding autowarefoundation/autoware#2056 to support the latest develop branch and some update.

**Comment by ChenxiTU** [comment]
Adding autowarefoundation/autoware#2046 to expand functions of rviz plugin - rosbag controller.

**Comment by kfunaoka** [comment]
Hi all and @aohsato @cirpue49 @sgermanserrano @gbiggs Sorry for the delay of releasing v1.11. I'd like to finalize pull request that can be merged to v1.11. Since I guess checking all pull request in this issue takes a long time, I'd like to discuss some of them can be merged or not in this issue, depending on their completion rate. I heard @aohsato and @cirpue49 are working hard to creating tests and documentation of Hybrid A* planner autowarefoundation/autoware#1594 and costmap_generator autowarefoundation/autoware#1774 for v1.11.

@aohsato @cirpue49 Would you approximate when these pull request can be ready to merge?

**Comment by k0suke-murakami** [comment]
Thank you for your consideration.
I already wrote tests for autowarefoundation/autoware#1774. Waiting for @sgermanserrano comment.
Not sure when will be merged. Guess 1~2 days.

**Comment by kfunaoka** [comment]
I heard some packages that depend on executing shell scripts such as autowarefoundation/autoware#2109 do not work fine because its execute permission is removed at copying files to install directory.

@hakuturu583 Does LGSVL simulator require the fix similar to autowarefoundation/autoware#2109?

**Comment by aohsato** [comment]
@kfunaoka @sgermanserrano 
Thanks for your discussion about releasing. In autowarefoundation/autoware#1594, I merged @sgermanserrano 's test code by autowarefoundation/autoware#1753 and added some fix to solve some test problems, so the library level unit test was completed.
Next, I tried to some test to nodes (10f93e3) but it's difficult to write without our standard code because there are mainly depended on pub/sub/tf system and complex perception results.
If possible, I want to make continuous changes based on the test code policy of v.1.12 or later.

**Comment by kfunaoka** [comment]
@aohsato We're now focusing on unit tests within not ROS level but library level. We can go forward to ROS level test and integration test after v1.11 or later. Once the CI in autowarefoundation/autoware#1594 is passed, would you declare that autowarefoundation/autoware#1594 was tested on real car, and then merge into develop?

**Comment by isamu-takagi** [comment]
@kfunaoka @hakuturu583 @yukkysaito 
Autoware Launcher in develop branch does not work with colcon.
Please review and merge the latest branch autowarefoundation/autoware#2056.

**Comment by kfunaoka** [comment]
For release note:

- [Fix] rosbag split size in runtime manager (#1817)
- [Fix] negative max_v values for negative omega (#1880)
- Merge pull request autowarefoundation/autoware#1815 from CPFL/feature/shape_estimation
- Feature/lgsvl sim configurator autowarefoundation/autoware#1893
- [Feature] negative estimate twist (#1798)
- [Fix] geo_pos_conv class member variable not initialized autowarefoundation/autoware#1913
- Feature/RosbagController (#1791)
- Changed Autoware/ros/run from being dependent on gksu to using sudo (#1918)
- Feature/autoware health checker (#1943)
- [Feature] waypoint replanner and extractor (#1951)
- [Feature] enable display errors (#1970)
- Fix/remove velodyne subtree (#1966)
- [fix] Install commands for all the packages (#1861)
- [feature] Baidu's CNN based LiDAR segmentation (#1800)
- Feature/spinnaker (#1984)
- Fix/colcon build (#2000)
- [Feature] Autoware Launcher (#1950)
- Feature/autoware health analyzer (#2004)
- [Feature] Improve AS vehicle interface (#1945)
- Merge pull request autowarefoundation/autoware#1930 from CPFL/feature/gazebo_simulator
- Merge pull request autowarefoundation/autoware#2020 from CPFL/fix/spinnaker
- Remove catkin_make_release* scripts (#2042)
- Update wiki demo (#2091)
- [Feature]PointPillars (#2029)
- Feature/rosbag_controller_split (#2046)
- [Feature] Rebuild decision maker (#1609)
- [Feature] costmap generator (#1774)
- Merge pull request autowarefoundation/autoware#2109 from CPFL/fix/gazebo_with_colcon
- Docker cleanup (#1946)
- Removing CUDA dependencies for Darknet Yolov3 (#1784)
- [Feature] Improve Hybrid A* planner (#1594)
- [fix] not predicting objects at x<=0 or y<=0 (#2122)

**Comment by kfunaoka** [comment]
Release Note

# Announcement
- Obsolete packages and nodes were removed as announced at [Ver.1.10.0](https://github.com/CPFL/Autoware/releases/tag/1.10.0) (#1812)
- Please use colcon for source build. Catkin is no longer supported (#2042)
The following commands are required once, when you build from source
```
sudo apt-get install -y python3-pip python3-setuptools
sudo pip3 install -U setuptools
sudo apt-get install -y python3-colcon-common-extensions
```

# New Features
- PointPillars: Fast Encoders for Object Detection from Point Clouds (#2029)
https://drive.google.com/file/d/1l6Grsg0JbBX5s7-JQ6eXQY169PCewJnW/view
- Baidu's CNN based LiDAR segmentation (#1800)
https://www.youtube.com/watch?v=fw7522gDBAc
- Shape/Pose Estimation based on PointCloud (#1815)
- Autoware Health Checker (#1943, autowarefoundation/autoware#1970, autowarefoundation/autoware#2004)
- ROS driver for pointgrey cameras using Spinnaker SDK (#1984)
- RosbagController (#1791, autowarefoundation/autoware#2046)

# Experimental Features
- Autoware Launcher provides simple easy-to-use user interface  (#1950)
- Decision maker is rebuilt based on real-world's requirements (#1609)

# Improvements
- Improved Hybrid A* (#1594) and Cost Map using prediction (#1774)
- Removed CUDA dependencies for Darknet (#1784)
- Improved AS vehicle interface (#1945)
- Gazebo Simulator can be launched from Runtime Manager (#1930)
- LVSVL Simulator can be launched from remote machine (#1893)
- Docker cleanup (#1946)
- Divided waypoint_loader into waypoint_loader, waypoin_replanner, and waypoint_extractor (#1951)

# Bug Fixes
- Fixed rosbag split size in runtime manager (#1817)
- Fixed max velocity on right turns at twist_filter (#1880)
- Fixed first topic of gnss_pose that kills ndt_matching (#1913)
- Fixed no prediction about objects at x<=0 or y<=0 (#2122)
- Inserted install commands for all the packages (#1861)
